### PR TITLE
RE2022-177: add WS env for WS downloader

### PR DESCRIPTION
**Pull request by Tianhao-Gu**

### CODE REVIEW

**Review by Xiangs18 — COMMENTED** [review]
LGTM for WS downloader.

Do we also need to update parsing script and NCBI downloader as well?

**Comment by Tianhao-Gu** [comment]
> LGTM for WS downloader.
> 
> Do we also need to update parsing script and NCBI downloader as well?

yes. I will update the parser and taskfarmer scheduler accordingly after this PR.

**Comment by MrCreosote** [comment]
Note that the NCBI downloader downloads from (obviously) NCBI, which doesn't have kbase environments, so it should be updated to just use `none` or something like that for an environment

**Comment by Tianhao-Gu** [comment]
> Note that the NCBI downloader downloads from (obviously) NCBI, which doesn't have kbase environments, so it should be updated to just use `none` or something like that for an environment

I don’t see any reason why we are creating an null environment for NCBI. I would just leave it as is.

**Comment by MrCreosote** [comment]
> I don’t see any reason why we are creating an null environment for NCBI. I would just leave it as is.

I guess for the `sourcedata` directory it doesn't matter since everything is reading from the `collectionssouce` directory, but there it seems like it'd make more sense to have the same directory structure as the workspace based collections

File: src/loaders/workspace_downloader/workspace_downloader.py [hunk 1]
```diff
@@ -18,8 +17,8 @@
   --source_version SOURCE_VERSION
                         Create a source version and link in data to that collection from the overall workspace source data dir
   --root_dir ROOT_DIR   Root directory. (default: /global/cfs/cdirs/kbase/collections)
-  --kb_base_url KB_BASE_URL
-                        KBase base URL, defaulting to prod (default: https://kbase.us/services/)
+  --env {CI,NEXT,APPDEV,PROD}
+                        KBase base URL, defaulting to prod (default: PROD)
```
[the remark this answers is not in the document]
Comment (Tianhao-Gu) [reply src/loaders/workspace_downloader/workspace_downloader.py]: 👍

File: src/loaders/workspace_downloader/workspace_downloader.py [hunk 2]
```diff
@@ -18,8 +17,8 @@
   --source_version SOURCE_VERSION
                         Create a source version and link in data to that collection from the overall workspace source data dir
   --root_dir ROOT_DIR   Root directory. (default: /global/cfs/cdirs/kbase/collections)
-  --kb_base_url KB_BASE_URL
```
[the remark this answers is not in the document]
Comment (Tianhao-Gu) [reply src/loaders/workspace_downloader/workspace_downloader.py:21]: we probably don't need to worry about it now. I will probably just mock the `KB_BASE_URL_MAP`.

File: src/loaders/workspace_downloader/workspace_downloader.py [hunk 3]
```diff
@@ -170,13 +176,13 @@ def _make_job_dir(root_dir, job_dir, username):
...
 def _make_collection_source_dir(
-        root_dir, collection_source_dir, collection, source_verion
+        root_dir, collection_source_dir, collection, source_verion, env
 ):
     """
     Helper function that creates a collection & source_version and link in data
     to that colleciton from the overall workspace source data dir.
     """
-    csd = os.path.join(root_dir, collection_source_dir, collection, source_verion)
+    csd = os.path.join(root_dir, collection_source_dir, collection, source_verion, env)
```
[the remark this answers is not in the document]
Comment (Tianhao-Gu) [reply src/loaders/workspace_downloader/workspace_downloader.py]: I moved the `ENV` to the top level. I honestly don't have a strong preference.

**Comment by Tianhao-Gu** [comment]
> 

👍

**Comment by Tianhao-Gu** [comment]
> > I don’t see any reason why we are creating an null environment for NCBI. I would just leave it as is.
> 
> I guess for the `sourcedata` directory it doesn't matter since everything is reading from the `collectionssouce` directory, but there it seems like it'd make more sense to have the same directory structure as the workspace based collections

👍

File: src/loaders/workspace_downloader/workspace_downloader.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (MrCreosote) [reply src/loaders/workspace_downloader/workspace_downloader.py:21]: :+1:

File: src/loaders/workspace_downloader/workspace_downloader.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (MrCreosote) [reply src/loaders/workspace_downloader/workspace_downloader.py]: :+1:

File: src/loaders/workspace_downloader/workspace_downloader.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (MrCreosote) [reply src/loaders/workspace_downloader/workspace_downloader.py]: :+1:

**Review by MrCreosote — APPROVED** [review]

**Pull request merged**